src/lib.rs

```diff
@@ -231,37 +292,39 @@ impl Sketch {
         if n > 2f64.powf(2f64.powf(f64::from(Q)) + f64::from(R)) {
             f64::INFINITY
         } else if n > 2f64.powf(f64::from(P) + 5.0) {
-            let d = (4.0 * n / m) / ((1.0 + n) / m).powi(2);
-            C * 2f64.powf(f64::from(P) - f64::from(R)) * d + 0.5
+            let s = n + m;
+            let ratio_factor = 4.0 * (n / s) * (m / s); // 4 n m / (n + m)^2
+            const SHIFT: u32 = P - (R as u32);
+            const K: f64 = C * ((1u64 << SHIFT) as f64); // precompute C * 2^(P-R)
+            K * ratio_factor
```

Comment:
I don't exactly get the reason for this change. The compiler will const-fold valid transformations